libvast/src/system/counter.cpp

```diff
@@ -52,16 +52,18 @@ void counter_state::init(expression expr, caf::actor index,
   behaviors_[collect_hits] = base.or_else(
     [this](table_slice_ptr slice) {
       // Construct a candidate checker if we don't have one for this type.
-      auto& checker = checkers_[slice->layout()];
-      if (caf::holds_alternative<caf::none_t>(checker)) {
+      auto it = checkers_.find(slice->layout());
+      if (it == checkers_.end()) {
         if (auto x = tailor(expr_, slice->layout())) {
-          checker = std::move(*x);
+          bool _;
+          std::tie(it, _) = checkers_.emplace(vast::record_type{slice->layout()}, std::move(*x));
```

Comment:
```suggestion
          std::tie(it, std::ignore) = checkers_.emplace(vast::record_type{slice->layout()}, std::move(*x));
```

Either that, or just:

```suggestion
          it = checkers_.emplace(vast::record_type{slice->layout()}, std::move(*x)).first;
```

Reply:
The latter one feels less noisy to me.